src/Initializer.js

```diff
@@ -1,289 +1,407 @@
 'use strict'
 
+var beautify = require('js-beautify').js_beautify
 /**
  * Initializer
  */
 class Initializer {
 
   /**
-   * Compile (static)
+   * constructor
    */
-  static compile (schema) {
-    let initializer = new Initializer(schema)
-    initializer.parse()
-    return initializer.compile()
+  constructor (schema, options) {
+    Object.assign(this, options || {})
```

Comment:
I bring this up largely because this project is so focused on optimization, although there are other reasons for doing this (explicitness, linting, IDE auto-complete support, etc) -- if at all possible, we should list and assign each individual option. So, instead of `Object.assign(this, options)`:

```js
constructor (schema, options = {}) {
  this.x = options.x
  this.y = options.y
  this.z = options.z
}
```
Part of the reason that it's a performance optimization, is that Node's V8 actually counts `this.whatever` assignments in the constructor, and structures the object accordingly. There's a good discussion of it her: http://mrale.ph/blog/2014/07/30/constructor-vs-objectcreate.html